TicketBooking/Controllers/BookingController.cs

```diff
@@ -0,0 +1,67 @@
+using Microsoft.AspNetCore.Mvc;
+using TicketBooking.Helpers;
+using TicketBooking.Models;
+
+namespace TicketBooking.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BookingController : ControllerBase
+    {
+        private readonly ILogger<BookingController> _logger;
+        private readonly ISeatLockProvider seatLockProvider;
+
+        public BookingController(ILogger<BookingController> logger, ISeatLockProvider seatLockProvider)
+        {
+            _logger = logger;
+            this.seatLockProvider = seatLockProvider;
+        }
+
+        [HttpGet]
+        public IActionResult Post(BookingRequest bookingRequest)
+        {
+            try
+            {
+                var customerData = CustomerDB.Get(bookingRequest.CustomerId);
+
+                if (bookingRequest.RequestDate == null)
+                    bookingRequest.RequestDate = DateTime.UtcNow;
+
+                this.seatLockProvider.LockSeat(bookingRequest.SeatNo);
+
+                IPaymentProvider payment= new CreditCardPaymentProvider();
+                var paymentStatus = payment.ChargeCustomer(bookingRequest.CustomerId, bookingRequest.Amount);
+
+                if (paymentStatus)
+                {
+                    Ticket ticketSaved = new Ticket();
+
+                    if (ticketSaved == null)
```

Comment:
can we remove the unreached code?. As per line number 37 ticketSaved object will not be null.